qlty-cloud/src/token/mod.rs

```diff
@@ -0,0 +1,104 @@
+mod auth_flow;
+
+use crate::Client;
+use anyhow::{Context, Result};
+use keyring::Entry;
+use tracing::warn;
+
+pub const SERVICE: &str = "qlty-cli";
+pub const DEFAULT_USER: &str = "default";
+
+pub struct Token {
+    pub user: String,
+}
+
+impl Default for Token {
+    fn default() -> Self {
+        Self::new(DEFAULT_USER)
+    }
+}
+
+impl Token {
+    pub fn new(user: &str) -> Self {
+        Self {
+            user: user.to_owned(),
+        }
+    }
+
+    pub fn get(&self) -> Result<String> {
+        self.entry()?.get_password().with_context(|| {
+            format!(
+                "Failed to get access token for service '{}' and user '{}'",
+                SERVICE, self.user
+            )
+        })
+    }
+
+    pub fn set(&self, token: &str) -> Result<()> {
+        self.entry()?.set_password(token).with_context(|| {
+            format!(
+                "Failed to set access token for service '{}' and user '{}'",
+                SERVICE, self.user
+            )
+        })
+    }
+
+    pub fn delete(&self) -> Result<()> {
+        self.entry()?.delete_credential().with_context(|| {
+            format!(
+                "Failed to delete access token for service '{}' and user '{}'",
+                SERVICE, self.user
+            )
+        })
+    }
+
+    fn entry(&self) -> Result<Entry> {
+        Entry::new(SERVICE, &self.user).with_context(|| {
+            format!(
+                "Failed to create keyring entry for service '{}' and user '{}'",
+                SERVICE, self.user
+            )
+        })
+    }
+}
+
+pub fn load_auth_token() -> Result<String> {
```

Comment:
Since this function will invoke the browser flow if a token is not available (which is a big impact) I would probably try to reflect that in the name. Maybe something like load_or_authorize_token.

Reply:
`auth_token` is the vocabulary itself for the token value, so `authorize_token` is kind of ambiguous in this context. `load_or_retreive_auth_token()`?